internal/resolver/dns/dns_resolver.go

```diff
@@ -130,6 +134,7 @@ func (b *dnsBuilder) Build(target resolver.Target, cc resolver.ClientConn, opts
 		cc:                   cc,
 		rn:                   make(chan struct{}, 1),
 		disableServiceConfig: opts.DisableServiceConfig,
+		pollTimer:            time.NewTimer(0),
```

Comment:
Actually, `pollTimer` doesn't need to be a field - you can make this a local in `watcher` and initialize it like this before the loop.  (Also, remember to create with `newTimer`.)

Reply:
Done.